CommonCompilerConfig.cmake

```diff
@@ -344,11 +353,10 @@ else()
                 set(optimization_flags_cc ${optimization_flags_cc} -mtune=generic)
         endif()
         if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU" AND NOT MINGW)
-                set(optimization_flags_cc ${optimization_flags_cc} -fuse-ld=gold)
                 # NOTE -flto must go together in both cc and ld flags; also, it's presently incompatible
                 # with the -g option in at least some GNU compilers (saw in `man gcc` on Ubuntu)
-                set(optimization_flags_cc ${optimization_flags_cc} -fuse-linker-plugin -flto-report -flto) #TODO fix LTO for clang
-                set(optimization_flags_lt ${optimization_flags_lt} -flto) #TODO LTO for clang too
+                set(optimization_flags_cc ${optimization_flags_cc} -fuse-linker-plugin -flto-report -flto -fno-fat-lto-objects) #TODO fix LTO for clang
+                set(optimization_flags_lt ${optimization_flags_lt} -flto -fno-fat-lto-objects) #TODO LTO for clang too
```

Comment:
OT: added support for LLD linker, if installed. And do "thin LTO" which should make linking phase faster.